tree/ntuplebrowse/v7/inc/ROOT/RBrowseVisitor.hxx

```diff
@@ -0,0 +1,94 @@
+/// \file ROOT/RBrowseVisitor.hxx
+/// \ingroup NTupleBrowse ROOT7
+/// \author Simon Leisibach <simon.satoshi.rene.leisibach@cern.ch>
+/// \date 2019-07-30
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RBrowseVisitor
+#define ROOT7_RBrowseVisitor
+
+#include <ROOT/RField.hxx>
+#include <ROOT/RFieldVisitor.hxx>
+
+
+class TBrowser;
+
+namespace ROOT {
+namespace Experimental {
+   class RNTupleBrowser;
+   
+   
+// Indicate the type of data a field contains. It is used to deduce if a histogram for that field can/should be displayed and also how.
+enum fieldDatatype {
```

Comment:
I think we should try to avoid defining another, parallel type enumeration besides the one already provided through the `RField` hierarchy.